lib/qmp_protocol.ml

```diff
@@ -20,8 +20,12 @@ let connect path =
   Printf.fprintf stderr "Connecting to %s\n%!" path;
   let sockaddr = Unix.ADDR_UNIX path in
   let s = Unix.socket Unix.PF_UNIX Unix.SOCK_STREAM 0 in
-  Unix.connect s sockaddr;
-  s, Unix.in_channel_of_descr s, Unix.out_channel_of_descr s
+  try
+    Unix.connect s sockaddr;
+    s, Unix.in_channel_of_descr s, Unix.out_channel_of_descr s
+  with e ->
+    Unix.close s;
+    raise e
```

Comment:
Shoulw we print something to stderr as done by negotiate or it is a pointless kind of log?

Reply:
We should print the error log.